client/dashboard/sites/index.tsx

```diff
@@ -248,11 +250,21 @@ export default function Sites() {
 		} ),
 		[ defaultView, viewOptions ]
 	);
-	const fields = useMemo(
-		() =>
-			hasA8CSites ? DEFAULT_FIELDS : DEFAULT_FIELDS.filter( ( field ) => field.id !== 'is_a8c' ),
-		[ hasA8CSites ]
-	);
+
+	const fields = useMemo( () => {
+		const excludedFields = new Set< string >();
```

Comment:
Doing the checks inside the filter predicate seems more readable than creating an intermediary set.

```ts
	const fields = useMemo(
		() =>
			DEFAULT_FIELDS.filter( ( field ) => {
				if ( field.id === 'is_a8c' && hasA8CSites ) {
					return false;
				}
				if ( field.id === 'icon.ico' && view.type === 'grid' ) {
					return false;
				}
				return true;
			} ),
		[ hasA8CSites, view.type ]
	);
```

But probably a matter of taste I guess.